docs/MinimallyImpactfulServicing.md

```diff
@@ -0,0 +1,99 @@
+NmrDeregister* being called from DriverUnload.
+
+eBPF Core exposes functions to eBPF Proxy to initiate and terminate the eBPF
+virtual machine as well as functions to dispatch messages to the virtual
+machine. During initiation, eBPF Core will deserialize any previously serialized
+state, recreating any eBPF objects. During termination, eBPF Core will serialize
+any eBPF objects.
+
+eBPF Proxy exposes function to eBPF Core to perform manipulation of IO Manager
+handles, specifically creation of new handles, closing handles, and resolving
+handles to eBPF object identifiers. In addition eBPF Proxy will expose functions
+to eBPF Core to read and write from a serialization stream.
+
+## Handle management
+eBPF for Windows exposes maps, programs, and links as Windows handles to user
+mode. All eBPF objects have a unique object ID assigned to them when they are
+created, with APIs to obtain handles to eBPF objects via there ID.
+
+eBPF Proxy registers with the Windows IO Manager as the driver that owns the
+\Device\EbpfIoDevice namespace. As a result, all handles created within this
+namespace are dispatched to the eBPF Proxy driver. The eBPF Proxy driver in turn
+dispatches these calls to the currently active instance of the eBPF Core driver
+via its dispatch table.
+
+Each IO Manager handle points to a Windows FILE_OBJECT structure. Within this
+structure, the device driver is responsible for maintaining and interpreting
+the both the FsContext and the FsContext2 fields. The eBPF Proxy driver stores
+the id of eBPF object that this file object points to in the FsContext2 field.
+
+Resolution of a IO Manager handle to a eBPF object is a two step process. The
+first step is to resolve the handle to the eBPF object identifier. The object
+identifier is then resolved to a pointer to the eBPF object.
+
+eBPF object identifier are maintained across serialization and as a result
+handles remain valid.
+
+## Serialization
+During initiation and termination of eBPF Core the eBPF Proxy will pass a
+pointer to an opaque stream context. eBPF Proxy will expose functions to iterate
+over a list of flat buffers store within eBPF Proxy.
+
+During initiation, eBPF Core will iterate over a collection of flat buffers
+stored in eBPF Proxy. For each buffer stored in eBPF Proxy, eBPF Core will
+recreate the eBPF object that was used to create the flat buffer. References
+between objects will be resolved from object ID to pointers to objects.
+
+During termination, eBPF Core will iterate over all eBPF objects. For each
+object eBPF core will construct a flat buffer with the representation of the
+state of that eBPF object and call a function in eBPF Proxy to store that flat
+buffer. References between objects will be converted from pointers to objects to
+object ID.
+
+Due to dependencies between eBPF objects, serialization will occur in the
+following order:
+1. Maps
+2. Programs
+3. Links
+
+Serialization format is still TBD, but possible candidates are:
```

Comment:
What about CBOR?  Much more compact than JSON, much easier to parse than ASN.1, and a standard used in many protocols, with open source libraries available such as `https://github.com/laurencelundblade/QCBOR`